src/main/java/algorithms/curated170/medium/parallelcourses/ParallelCoursesDFS.java

```diff
@@ -1,36 +1,49 @@
 package algorithms.curated170.medium.parallelcourses;
 
 import java.util.ArrayList;
-import java.util.HashMap;
+import java.util.LinkedList;
 import java.util.List;
 
 public class ParallelCoursesDFS {
 
     private int maxPathLength = Integer.MIN_VALUE;
     private boolean hasCycle = false;
-    HashMap<Integer, List<Integer>> graph;
+    List<List<Integer>> courseChiMap;
 
     boolean[] visited;
     int[] depth;
 
     public int minimumSemesters(int n, int[][] relations) {
-        graph = new HashMap<>();
+
+        createMap(n, relations);
+
+        searchNodes(n);
+
+        return hasCycle ? -1 : maxPathLength;
+    }
+
+    private void createMap(int n, int[][] relations) {
+        courseChiMap = new LinkedList<>();
```

Comment:
We should also change this to:
 courseChiMap = new ArrayList<>();
 
 Otherwise it will take 80ms instead of 5ms.

Reply:
Added.